test/crate_type/CMakeLists.txt

```diff
@@ -0,0 +1,6 @@
+corrosion_tests_add_test(crate_type "cpp-exe")
```

Comment:
It seems like the test is currently not added. You have to add the following in the `tests` directory CMakeLists.txt.
```cmake
add_subdirectory(crate_type)
```
Since `--crate-type=` is used in the test and is a rather new cargo flag, the test also should only be conditionally added if the Rust version is new enough.